### ISSUE TITLE
Adding New Cluster Always Selects Flannel Network Provider

### ISSUE THREAD

**Issue by ekristen** [rancher/rancher#18882]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):** 

1. Add custom cluster 
2. Select any provider other than Flannel
3. Upon clicking next

**Result:**

When you click save it always reverts to Flannel. I'm unable to use any other provider which has made testing ARM64 tough :(

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.2.0-rc4
- Installation option (single install/HA): HA

**Comment by cjellick** [comment]
@rmweir can you figure out if this is a backend or frontend issue?

**Comment by ekristen** [comment]
I can confirm that when I hit save in the UI, the UI greys out fields and switches the dropdown back to Flannel and then proceeds to the next screen. I'm trying to verify via the API that it indeed is saving it as Flannel.

**Comment by ekristen** [comment]
![provider-switch mov](https://user-images.githubusercontent.com/48329/54314350-0d7f2b80-45a1-11e9-85d6-3e6a5f797f79.gif)

**Comment by westlywright** [comment]
The UI is reseting that field after the resource saves but if you view the chrome network logs (or view the resource in api) the data for `rancherKubernetesEngineConfig.network.plugin` is correct. Normally the you shouldn't even see that reset but something is delaying the progress to the next page. You can verify data is correct in the API UI by selecting the `View In API` option from the action menu for the cluster. I do see that `flannel_backend_type` is always  set in the options but the API should drop this field if the correct plugin isn't being used.

**Comment by ekristen** [comment]
Full disclosure, I was on alpha3 at one point, but I *thought* I had cleared everything out when I re-installed rc1, it appears I missed something. 

I can definitely state that if you start with alpha3 -> rc1 -> rc2 -> rc3 -> rc4 you will see this behavior.

I'm running through all the rc upgrades right now from scratch just to be sure, but this could very well be a latent issue with alpha3.

Yes I know alpha3 upgrade aren't supported, wasn't meaning for that to come in as I thought I had wiped that out.
